pkgs/tools/nix/nixos-option/src/meson.build

```diff
@@ -0,0 +1,32 @@
+src = [
+  'nixos-option.cc',
+  'libnix-copy-paste.cc',
+]
+
+cc = meson.get_compiler('cpp')
+
+autotool_config = configuration_data()
+# nix defines this with autotools
+if cc.has_member('struct dirent', 'd_type', prefix: '#include <dirent.h>')
+  autotool_config.set('HAVE_STRUCT_DIRENT_D_TYPE', 1)
+endif
```

Comment:
I think if you `-include` the right `config.h`, it is fine. That should also be made less hacky upstream.

Reply:
See https://github.com/NixOS/hydra/blob/f974891c76e295240017dd7f04d50ecb4b70284e/configure.ac#L56